Backend/src/main/java/idealab/api/operations/PrintJobOperations.java

```diff
@@ -176,6 +209,35 @@ public GenericResponse updatePrintJobStatus(Integer printId, PrintJobUpdateReque
 
         if(employee == null || printJob == null){
             ErrorType.PRINT_JOB_UPDATE_FAILED.throwException();
+        } 
+
+        // If the status is changed to final, delete the print job rank from the rank table
+        // and check if another one needs to be added
+        if (requestStatus == Status.REJECTED || requestStatus == Status.COMPLETED) {
+            // Delete rank from queue table
+            Queue deleteQueue = queueRepo.findByPrintJobId(printJob);
+            queueRepo.delete(deleteQueue);
+
+            // Get list of print jobs for customer + remove any completed/rejected
+            // + get oldest date in list
+            CustomerInfo customer = printJob.getCustomerInfo();
+            List<PrintJob> printJobs = printJobRepo.findByCustomerInfo(customer);
+            printJobs.removeIf(job -> (job.getStatus() == Status.REJECTED || job.getStatus() == Status.COMPLETED));
+
+            if (printJobs.size() > 1) {
```

Comment:
I would rather see the PrintJobs class implement Comparable interface, I think it would make the code cleaner.

Reply:
Done!